### PR TITLE
More documentation fixes

### PR DESCRIPTION
**Pull request by adigitoleo**
- Make documentation of CRSS more precise
- Add missing enum docstrings
- Promote comments to docstrings in core tests
- Make print statements into logging calls

### CODE REVIEW

File: src/pydrex/minerals.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
-"""> PyDRex: Lagrangian data structures and fabric/CRSS selection.
+"""> PyDRex: Lagrangian data structures and mineral fabric definitions.
 
-Acronyms:
+**Acronyms:**
 - CRSS = Critical Resolved Shear Stress,
-    i.e. components of stress acting on each slip system in the grain reference frame
+    i.e. threshold stress required to initiate slip on a slip system,
+    normalised to the stress required to initiate slip on the softes slip system
```
Comment (Patol75) [inline src/pydrex/minerals.py:6]: softest

**Review by Patol75 — CHANGES_REQUESTED** [review]
Found one typo, the rest looks good!

**Review by Patol75 — APPROVED** [review]

**Comment by Patol75** [comment]
It seems like we have a test failure. Any merit in replacing some of these `assert` statements by [numpy.testing](https://numpy.org/doc/stable/reference/routines.testing.html) functions to get more information on failures? `np.testing.assert_allclose()` should be a good replacement for `assert np.isclose()`.

**Comment by adigitoleo** [comment]
> test failure

I've edited the tolerance value and will submit that in the next test fixes. Apologies, I re-ran the job before reading your comment. The failure was once again due to one of the output diagnostics being slightly different and slightly past the tolerance.

There are two sources of randomness in the outputs:
- Random orientations in initial condition are not deterministic. Maybe I should seed the RNG for the tests.
- The diagnostics (M-index, angle) rely on re-sampled orientations, which once again uses an RNG. Maybe I should also seed that one for now, ultimately I would like for it to converge a bit better and am testing now with different number of grains, timesteps, etc.

> replacing some of these `assert` statements by [numpy.testing](https://numpy.org/doc/stable/reference/routines.testing.html) functions

I guess you mean using assert_allclose against the entire `misorient_angles`/`misorient_indices` arrays, instead of just assert statements against specific values? Currently that would make the test even more flaky, but it's probably a better approach once I seed the RNGs.

**Pull request merged**